src/ExactDiagonalization/init_and_solvers.jl

```diff
@@ -13,27 +13,17 @@ function CommonSolve.init( # no algorithm specified as positional argument
     kwargs...
 )
     kwargs = (; prob.kwargs..., kwargs...) # remove duplicates
-    algorithm = get(kwargs, :algorithm, LinearAlgebraSolver())
-    kwargs = delete(kwargs, :algorithm)
-    new_prob = ExactDiagonalizationProblem(prob.hamiltonian, prob.initial_vector; kwargs...)
+    algorithm = get(kwargs, :algorithm, prob.algorithm)
+    info = get(kwargs, :info, false)
+    warn = get(kwargs, :warn, false)
```

Comment:
```suggestion
    warn = get(kwargs, :warn, true)
```
The docstring says `warn` defaults to `true`.

Reply:
Here a new `ExactDiagonalizationProblem` is created from an old one. I figured that default warning messages should only be emitted on the first creation of an instance and that people who call `init` or `solve` with the algorithm as a positional argument know what they are doing.